lib/sql.ts

```diff
@@ -259,12 +263,154 @@ export class SQLDatabaseManager extends QueryProcessManager<DatabaseQuery, any>
 	}
 }
 
+export const tables = new Map<string, DatabaseTable<any>>();
+
+export class DatabaseTable<T> {
+	database: SQLDatabaseManager;
+	name: string;
+	primaryKeyName: string;
+	constructor(
+		name: string,
+		primaryKeyName: string,
+		database: SQLDatabaseManager
+	) {
+		this.name = name;
+		this.database = database;
+		this.primaryKeyName = primaryKeyName;
+		tables.set(this.name, this);
+	}
+	private SQL: typeof import('sql-template-strings').SQL = (() => {
+		try {
+			return require('sql-template-strings');
+		} catch {
+			return () => {
+				throw new Error("Using SQL-template-strings without it installed");
+			};
+		}
+	})();
+	async selectOne<R = T>(
+		entries: string | string[],
+		where?: SQLStatement
+	): Promise<R | null> {
+		const query = where || this.SQL``;
+		query.append(' LIMIT 1');
+		const rows = await this.selectAll<R>(entries, query);
+		return rows?.[0] || null;
+	}
+	selectAll<R = T>(
+		entries: string | string[],
+		where?: SQLStatement
+	): Promise<R[]> {
+		const query = this.SQL`SELECT `;
+		if (typeof entries === 'string') {
+			query.append(' * ');
+		} else {
+			for (let i = 0; i < entries.length; i++) {
+				query.append(entries[i]);
+				if (typeof entries[i + 1] !== 'undefined') query.append(', ');
+			}
+			query.append(' ');
+		}
+		query.append(`FROM ${this.name} `);
+		if (where) {
+			query.append(' WHERE ');
+			query.append(where);
+		}
+		return this.all<R>(query);
+	}
+	get(entries: string | string[], keyId: SQLInput) {
+		const query = this.SQL``;
+		query.append(this.primaryKeyName);
+		query.append(this.SQL` = ${keyId}`);
+		return this.selectOne(entries, query);
+	}
+	updateAll(toParams: Partial<T>, where?: SQLStatement, limit?: number) {
+		const to = Object.entries(toParams);
+		const query = this.SQL`UPDATE `;
+		query.append(this.name + ' SET ');
+		for (let i = 0; i < to.length; i++) {
+			const [k, v] = to[i];
+			query.append(`${k} = `);
```

Comment:
This doesn't sanitize column names, which is fine, but something to be careful of (I need to remember it for future reviews which is half the reason I'm commenting it here).